File: blood_bank_PyTest/bloodbank.py
from donor import Donor


class BloodBank:
    def __init__(self):
        self.all_my_donors = []

    def add_donor(self, new_national_id, new_first_name, new_last_name, new_gender):
        new_donor = Donor(new_national_id, new_first_name, new_last_name, new_gender)
        self.all_my_donors.append(new_donor)

    def find_donor(self, target_donor_id):
        return next((donor for donor in self.all_my_donors if donor.id == target_donor_id), None)

    def sort_donor(self):
        self.all_my_donors = sorted(self.all_my_donors, key=lambda donor: donor.id)

    def display_donor(self):
        self.sort_donor()
        result = ''
        for aDonor in self.all_my_donors:
            result += str(aDonor) + '\n'
        return result

    def displayPositiveDonations(self):
        self.sort_donor()
        result = ''
        for aDonor in self.all_my_donors:
            if aDonor.has_positive_bloodgroup():
                result += str(aDonor) + '\n'
                result += aDonor.get_donations()
        return result

File: Project_Files/Tests/JavaScript_Bloodbank_Python_SocialClub/model_answer/MochaChai/src/BloodBank.js
class BloodBank {
  constructor () {
    this.allMyDonors = []
  }
  addDonor (newNationalId, newFirstName, newLastName, newGender) {
    var aDonor = new Donor(newNationalId, newFirstName, newLastName, newGender)
    this.allMyDonors.push(aDonor)
  }
  findDonor (targetNationalId) {
    return this.allMyDonors.find(aDonor => aDonor.nationalId === targetNationalId)
  }
  sortDonors () {
    this.allMyDonors.sort(function (a,b) {return a.nationalId-b.nationalId})
  }
  displayDonors () {
    this.sortDonors()
    let result = ''
    for (let aDonor of this.allMyDonors) {
      result += aDonor + '\n'
    }
    return result
  }
  displayPositiveDonations () {
    this.sortDonors();
    let result ='';
    for (let aDonor of this.allMyDonors) {
      if(aDonor.hasPositiveBloodGroup()) {
        result += aDonor + '\n'
        result += aDonor.getDonations()
      }
      // if(aDonor.hasPositiveBloodGroup()) {
      //   aDonor.sortDonations()
      //   result += aDonor.getDonations()
      // }
    }
    return result
  }
}
